bundledatasrc_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package charm
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/juju/testing"
+	gc "gopkg.in/check.v1"
+)
+
+type BundleDataSourceSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&BundleDataSourceSuite{})
+
+func (s *BundleDataSourceSuite) TestReadMultiDocBundle(c *gc.C) {
+	path := bundleDirPath(c, "wordpress-multidoc")
+	ds := FileBundleDataSource(filepath.Join(path, "bundle.yaml"))
+
+	parts, err := ds.read()
+	c.Assert(err, gc.IsNil)
+	c.Assert(parts, gc.HasLen, 3)
+	s.assertFieldPresent(c, parts[1], "applications.wordpress.offers.offer1.endpoints")
+	s.assertFieldPresent(c, parts[2], "applications.wordpress.offers.offer1.acl.admin")
```

Comment:
don't you want to both assert the positive, and the negative? (assertFieldNotPresent) ?

Reply:
Actually, there isn't any value in testing the negative as the yaml parser will simply not populate the map if the keys do not exist.